### PR TITLE
Implemented a Go consumer

### PR DESCRIPTION
**Pull request by dim**
@mxmCherry

### CODE REVIEW

File: consumer.go [hunk 1]
```diff
@@ -24,8 +24,9 @@ type ConsumerOptions struct {
 	// Default: auto-detected from URL path.
 	Compression Compression
 
-	// OnError callback is triggered on sync errors.
-	OnError func(error)
+	// AfterSync callbacks are triggered after each sync, receiving
+	// the updated status and error (if occurred).
+	AfterSync func(updated bool, bool error)
```
Comment (mxmCherry) [inline consumer.go]: mistype - you meant "bool error"-> "err error"?

Comment (mxmCherry) [reply consumer.go]: Also, can it ever be `true, NonNilErr`?

Comment (dim) [reply consumer.go]: It returns:

* `true, nil` if the sync has updated the data
* `false, nil` if the sync has run but not updated any data (not modified)
* `false, err` if there was an error

Will fix typo

File: consumer.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+	// parse feed
+	data, size, err := f.parse(d)
+	if err != nil {
+		return err
+	}
+
+	// update stores
+	f.data.Store(data)
+	atomic.StoreInt64(&f.size, size)
+	atomic.StoreInt64(&f.lastModMs, msec)
```
[the remark this answers is not in the document]
Comment (dim) [reply consumer.go]: `lastModMs` is read in `LastModified() time.Time`

File: consumer_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		c, err := feedx.GZipCompression.NewWriter(w)
+		Expect(err).NotTo(HaveOccurred())
+		defer c.Close()
+
+		f, err := feedx.JSONFormat.NewEncoder(c)
+		Expect(err).NotTo(HaveOccurred())
+		defer f.Close()
+
+		Expect(f.Encode(msg)).To(Succeed())
+		Expect(f.Encode(msg)).To(Succeed())
```
[the remark this answers is not in the document]
Comment (dim) [reply consumer_test.go]: I will be simpler once I add `Producer`, until then, it's a bit verbose

File: consumer.go (hunk 1 above)
Comment (mxmCherry) [reply consumer.go]: I'd just have `ErrNotModified` for second case, probably

Comment (dim) [reply consumer.go]: @mxmCherry preferences ...

**Pull request merged**